app/code/Magento/Catalog/Model/ProductRepository.php

```diff
@@ -326,6 +326,10 @@ private function cacheProduct($cacheKey, \Magento\Catalog\Api\Data\ProductInterf
      */
     protected function initializeProductData(array $productData, $createNew)
     {
+        if (empty($productData['type_id'])) {
+            $productData['type_id'] = \Magento\Catalog\Model\Product\Type::DEFAULT_TYPE;
```

Comment:
Hi @gomencal,

So, we don't have any kind of schema validation for WebAPI, right? Like a way to forbid empty type_id.

Where exactly error occurs in case of empty type_id? Maybe it's better to put such assignment there instead of repository?